From e8dcb703f0e7539f4f8d0df285ac83e9a19de064 Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Mon, 12 Jan 2026 04:40:09 +0800
Subject: [PATCH 10/21] fixes fd leak in exp_command.c

---
 exp_command.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/exp_command.c b/exp_command.c
index b554b18..c055ffd 100644
--- a/exp_command.c
+++ b/exp_command.c
@@ -1161,7 +1161,7 @@ Exp_SpawnObjCmd(
     /* if stty finds dev(stderr) != dev(stdout) */
 
     /* save error fd while we're setting up new one */
-    errorfd = fcntl(2,F_DUPFD,3);
+    errorfd = fcntl(2,F_DUPFD_CLOEXEC,3);
     /* and here is the macro to restore it */
 #define restore_error_fd {close(2);fcntl(errorfd,F_DUPFD,2);}
 
-- 
2.52.0.windows.1

