From ecd58f5c85ede4ebd9878f40bff61b8de1c8bb94 Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Mon, 12 Jan 2026 04:43:34 +0800
Subject: [PATCH 15/21] mkpasswd.man update

---
 example/mkpasswd.man | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/example/mkpasswd.man b/example/mkpasswd.man
index 9da760c..cf0c12d 100644
--- a/example/mkpasswd.man
+++ b/example/mkpasswd.man
@@ -40,8 +40,8 @@ The following example creates a 20 character password.
 
 The
 .B \-d
-flag defines the minimum number of digits that must be in the password.
-The default is 2.  The following example creates a password with at least
+flag defines the number of digits that must be in the password.
+The default is 2.  The following example creates a password with
 3 digits.
 
 	mkpasswd -d 3
@@ -53,12 +53,12 @@ The default is 2.
 
 The
 .B \-C
-flag defines the minimum number of uppercase alphabetic characters that must be in the password.
+flag defines the number of uppercase alphabetic characters that must be in the password.
 The default is 2.
 
 The
 .B \-s
-flag defines the minimum number of special characters that must be in the password.
+flag defines the number of special characters that must be in the password.
 The default is 1.
 
 The
@@ -80,7 +80,7 @@ By default, it is suppressed.
 
 .SH EXAMPLE
 The following example creates a 15-character password
-that contains at least 3 digits and 5 uppercase characters.
+that contains 3 digits and 5 uppercase characters.
 
 	mkpasswd -l 15 -d 3 -C 5
 
-- 
2.52.0.windows.1

