From 5eef60a2de65dc80fa9976c2621a67459bdb104f Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Tue, 6 Jan 2026 08:49:07 +0800
Subject: [PATCH] Link zlib to _cygwin_dll_entry

---
 Makefile.in        | 2 +-
 win32/Makefile.gcc | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index cb8b00a..27141dd 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -275,7 +275,7 @@ gzwrite.lo: $(SRCDIR)gzwrite.c
 
 
 placebo $(SHAREDLIBV): $(PIC_OBJS) libz.a
-	$(LDSHARED) $(SFLAGS) -o $@ $(PIC_OBJS) $(LDSHAREDLIBC) $(LDFLAGS)
+	$(LDSHARED) $(SFLAGS) -o $@ $(PIC_OBJS) $(LDSHAREDLIBC) $(LDFLAGS) -Wl,-e -Wl,_cygwin_dll_entry
 	rm -f $(SHAREDLIB) $(SHAREDLIBM)
 	ln -s $@ $(SHAREDLIB)
 	ln -s $@ $(SHAREDLIBM)
diff --git a/win32/Makefile.gcc b/win32/Makefile.gcc
index 081e391..f63053d 100644
--- a/win32/Makefile.gcc
+++ b/win32/Makefile.gcc
@@ -88,7 +88,7 @@ $(STATICLIB): $(OBJS) $(OBJA)
 $(IMPLIB): $(SHAREDLIB)
 
 $(SHAREDLIB): win32/zlib.def $(OBJS) $(OBJA) zlibrc.o
-	$(CC) -shared -Wl,--out-implib,$(IMPLIB) $(LDFLAGS) \
+	$(CC) -shared -Wl,-e -Wl,_cygwin_dll_entry -Wl,--out-implib,$(IMPLIB) $(LDFLAGS) \
 	-o $@ win32/zlib.def $(OBJS) $(OBJA) zlibrc.o
 	$(STRIP) $@
 
-- 
2.52.0.windows.1

